sys/net/application_layer/gcoap/gcoap.c

```diff
@@ -769,12 +675,68 @@ int gcoap_req_init(coap_pkt_t *pdu, uint8_t *buf, size_t len,
 
 ssize_t gcoap_finish(coap_pkt_t *pdu, size_t payload_len, unsigned format)
 {
-    /* reconstruct full PDU buffer length */
-    size_t len = pdu->payload_len + (pdu->payload - (uint8_t *)pdu->hdr);
+    unsigned format_offset = UINT_MAX;
```

Comment:
The use of this variable confuses me. It looks to me as if it is only used to remember if there was a content-format option

Reply:
Yes, that's right. I will generally add more comments to make the logic in gcoap_finish() clearer. This tortured nature of this code is why I plan to deprecate gcoap_finish() in favor of gcoap_opt_format_finish(), which is called after all options have been entered but before the payload has been written. See [API Options](https://github.com/RIOT-OS/RIOT/wiki/CoAP-API-Options-Update) page for details.